squad_client/core/models.py

```diff
@@ -402,6 +408,15 @@ def submit_results(self):
             log=self.log,
             attachments=self.attachments)
 
+    __status__ = None
+
+    def status(self, count=ALL, **filters):
+        if self.__status__ is None:
+            TestRunStatus.endpoint = '%s%d/status' % (self.endpoint, self.id)
+            filters['suite__isnull'] = 'true'
```

Comment:
I think this should be optional. At least it should be possible to overwrite it by passing filters as parameter.

Reply:
So if there is any 'suite__isnull' we should use it, and if not present then we set it to true?

Reply:
My point is that you should be able to filter any suite, not only 'null' one.

Reply:
@chaws shouldn't we rid off the lines in bucket_metric_and_test_suites that tally the tests_pass/fail...etc? same for the tests_pass attr in the class, should we leave those or remove them since we can fetch status on the fly now?

Reply:
@amrohassaan I thought about that too, but thinking performance-wise, if tests are already there, we might as well just calculate status out of it instead of making a new trip to the api. In case tests aren't there then we should use status. What do you think?

Reply:
@chaws I see your point, not sure if we will still need the bucketing func at all, since most report requestors said they don't care about testruns. Recently Ryan Harkin too. Will there be a case that still need this?

Reply:
hmm, honestly I don't mind removing it. I just thought there could be an easy way to fetch all results for a given build.